streams/actions/src/main/java/com/hortonworks/streamline/streams/actions/topology/service/TopologyActionsService.java

```diff
@@ -152,15 +148,39 @@ public boolean killTestRunTopology(Topology topology, TopologyTestRunHistory his
     }
 
     public void killTopology(Topology topology, String asUser) throws Exception {
-        getTopologyContext(topology, asUser).kill();
+        mutateTopologyContext(TopologyContext::kill, topology, asUser);
     }
 
     public void suspendTopology(Topology topology, String asUser) throws Exception {
-        getTopologyContext(topology, asUser).suspend();
+        mutateTopologyContext(TopologyContext::suspend, topology, asUser);
     }
 
     public void resumeTopology(Topology topology, String asUser) throws Exception {
-        getTopologyContext(topology, asUser).resume();
+        mutateTopologyContext(TopologyContext::resume, topology, asUser);
+    }
+
+    @FunctionalInterface
+    interface ConsumerWithThrowsException<T> {
+        void accept(T t) throws Exception;
+    }
+
+    private void mutateTopologyContext(ConsumerWithThrowsException<TopologyContext> mutatingFunction,
+                                       TopologyContext topologyContext) throws Exception {
+        try {
+            transactionManager.beginTransaction(TransactionIsolation.DEFAULT);
+            mutatingFunction.accept(topologyContext);
+        } catch (IgnoreTransactionRollbackException e) {
+            transactionManager.commitTransaction();
+            throw convertThrowableToException(e.getCause());
```

Comment:
you already invoke the `getCause` in `convertThrowableToException` so shouldnt this be convertThrowableToException(e) ?

Reply:
Yeah I realized `convertThrowableException` should handle passed exception instead of caused exception in passed exception, given that there's no context for the behavior. I'll leave this as is and fix `convertThrowableException` to handle passed exception.